core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexUpgradeService.java

```diff
@@ -112,9 +116,25 @@ private void checkMappingsCompatibility(IndexMetaData indexMetaData) {
             // been started yet. However, we don't really need real analyzers at this stage - so we can fake it
             IndexSettings indexSettings = new IndexSettings(indexMetaData, this.settings);
             SimilarityService similarityService = new SimilarityService(indexSettings, Collections.emptyMap());
+            final NamedAnalyzer fakeDefault = new NamedAnalyzer("fake_default", new Analyzer() {
+                @Override
+                protected TokenStreamComponents createComponents(String fieldName) {
+                    throw new UnsupportedOperationException("shouldn't be here");
+                }
+            });
+            final Map<String, NamedAnalyzer> analyzerMap =    new AbstractMap<String, NamedAnalyzer>() {
+                @Override
+                public NamedAnalyzer get(Object key) {
+                    return new NamedAnalyzer((String)key, fakeDefault.analyzer());
```

Comment:
It is probably worth adding a comment about why we need to support doing this for *any* string. This map's `entrySet` method doesn't agree with the `get` which is probably fine for this but we should explain it.